### ISSUE TITLE
WCAG2 Compliance?

### ISSUE THREAD

**Issue by frastlin** [odoo/odoo#13950]
# Impacted versions:

all
# Steps to reproduce:

Visit any Odoo page on any app and very few have `<h1>` or any headings. There are also lots of unlabeled buttons, especially when choosing the different views for bank accounts and whatnot. I believe there is a list view, pivot table view and graph view and the buttons here are not labeled. Also, there is a silly bug in the HTML of `<input type="text">` tags where they are all missing an ID. The label above them needs to have the from tag match an id in the input field.
On:
https://www.odoo.com/page/contactus

The "send" button is also a clickable element and it should either be a `<input type="submit" value="Send">,` or `<button>Send</button>.`

I also didn't see any way for low vision users to change the contrast or size of the text.

When entering text in an edit field, the auto complete options would not read when I arrow up and down with the screen reader. There was a "4 options" announcement, but the reading of the options didn't work. The options should have announced as I pressed the up and down arrows.

For testing, I used the latest stable version of Firefox and the latest stable version of NVDA:
http://www.nvaccess.org/

I was also using the accounting app of Odoo.
# Importance of WCAG2 compliance and accessibility:

Being WCAG2 compliant is very important, especially if Odoo wishes to become a serious option in the U.S. education or Governmental sector:
http://webaim.org/articles/laws/usa/
Also, with the pending laws from the UN Convention on the Rights of Persons with Disabilities such as the European Accessibility Act:
http://eur-lex.europa.eu/legal-content/EN/TXT/?uri=COM:2015:0615:FIN

Accessibility will start becoming a huge factor in deciding what systems a business in the EU or U.S. will implement.

This bug report, in fact, comes directly out of an evaluation of open-source ERP systems on accessibility for my company. Currently none of the ERPs I tried were completely accessible and Odoo would be the ERP we would choose if we didn't need accessibility. Odoo is very close to being accessible, it just needs a few modifications to be compliant.
# Previous Requests for WCAG2:

There was a thread that was posted asking about WCAG2 compliance at:
https://www.odoo.com/forum/help-1/question/odoo-and-wcag-compliance-83096

But there was no response.
# Odoo's Cooperation?

How open is the development team on making WCAG2 compliance a focus? I would be willing to help, but I don't want to spend a lot of time on something if there is little if no support from the core development team.

**Comment by JKE-be** [comment]
not a priority currently

Thanks for suggestion.

**Comment by monfero** [comment]
@frastlin , your post is three years old so, do you know if Odoo have improved their accesibility? Do you know other ERPs WCAG compliant?

**Comment by frastlin** [comment]
I haven't heard anything and I have not found any ERP or PSA with an accessibility team.
Both Salesforce and Intuit have great accessibility teams, so we have been using those two products and forgoing an ERP.
The fact all these ERPs are not accessible is an incredible problem that is closing the workplace to millions of people. Dominoes pizza is paying lots of money now because their platform was not accessible to their customer. I would like to know, since I'm the customer of something like Odoo or a not open-source ERP, if the same law still applies.
I'll be talking to some lawyers this week or next to see if this is something that can be settled one way or another. It's dumb that programmers are not forced to use good practice and HTML when every page on the web is in html.

**Comment by JKE-be** [comment]
~2 year ago, a big work has been done to make odoo compatible with some readers.

Meanwhile not sure that we continue the work on each new component. But is should be a lot better than in 2016.

**Comment by FredrikArvas** [comment]
Hi
I am working in a team with two large Odoo CE 12.0 projects in the public sector in Sweden/Europe. We need to follow the WCAG2 / A11y regulations and we want to team up with to get this solved. 

The current issues we need to fix are mainly related to screenreaders (for blind users) where we have blocking issues with: 
- kanban-views
- tab-order
- the main and sub-navigation 

Let´s start a project and fix this for CE 12.0. 
or is it solved in v13 or v14?
/Fredrik

**Comment by Yenthe666** [comment]
@FredrikArvas well, nothing related to this can be changed in stable versions as it is against the stable policy. You could make PR's against the upcoming V15 (now master) to improve these things so that future versions get better but that's about it.

**Comment by FredrikArvas** [comment]
Thank you @Yenthe666 - This is good for the future, but it will not help us during 2021. :-(
Since we are working on behalf "of the citizens" we do not want to keep them waiting. 

Since I have not worked with community-projects with Odoo before, could you give me any advice forward?
both short term (v12, v13 etc...) and longterm (v15).

**Comment by Yenthe666** [comment]
Your best bet is probably to add custom modules on top of default Odoo to implement the changes you need (as you did with the linked modules). Afterwards if you have time and feel like it you could make PR's to add this to the future V15 and then somebody from Odoo can take a look at it :)
For the versions 12 to 14 you will have to use your own solutions and forward port your custom modules though.

**Comment by kbb01** [comment]
Hi Does anyone know if odoo v15 is now compliant? Thanks

**Comment by fredrikarvas-vertel** [comment]
We have worked with some testing and the v15 is more compliant then previous versions and especially the portal-interface is better. But there are still a lot of work to be done on the "Odoo inside" (internal users). The main problem is with screen readers, there are several severe issues with the navigation. Fabien (CEO of Odoo) said on the Odoo Experience 2021 that they had an employee who used a screeen reader (typically used by blind users) and she said Odoo was a lot easier than other ERP-systems. But we still need to increase the usability. I have two governement clients who want to contribute. Please join us!
